src/Equinox.Core/CachingDecorator.fs

```diff
@@ -0,0 +1,51 @@
+namespace Equinox.Core
+
+open Serilog
+open System
+open System.Threading
+open System.Threading.Tasks
+
+type IReloadableCategory<'event, 'state, 'context> =
+    inherit ICategory<'event, 'state, 'context>
+
+    abstract Reload: log: ILogger * streamName: string * requireLeader: bool * streamToken: StreamToken * state: 'state * ct: CancellationToken
+                     -> Task<struct (StreamToken * 'state)>
+
+type CachingDecorator<'event, 'state, 'context>
```

Comment:
This is a great change! Did you notice any deviations between the "caching category" implementations when extracting this, or was this very clean cut?

Reply:
While there were once some minor arbitrary variations, I had been gently nudging them to convergence over time, so the high level logic was identical